### PR TITLE
Fix `tap()` not being able to tap on widget previously found by `scrollTo()`

### LINKED ISSUE #1046: Can't tap on widget previously found by `scrollTo()`
Code can be found here: https://github.com/jBorkowska/flutter-wonderous-app/blob/patrol-bug/integration_test/test_test.dart#L25
In this line, we scroll to a widget and then try to tap on it. Scrolling works fine, but tap throws an error:

Image

Possible reason: there are 2 widgets with the same key in the widget tree at this moment in test: a container in https://github.com/jBorkowska/flutter-wonderous-app/blob/patrol-bug/lib/ui/screens/home/wonders_home_screen.dart#L293 and a stack in https://github.com/jBorkowska/flutter-wonderous-app/blob/patrol-bug/lib/ui/wonder_illustrations/common/animated_clouds.dart#L100. The container is found by scrollTo, but in error message from tap only the stack is mentioned (as offscreen). Also stack is higher in the widget tree than container, which means it's analyzed before container when resolving the finder. That potentially means, that scrollTo analyzed stack, but ignored it as offstage, but tap took first one from the tree and then checked hitTestable(), which returned false, resulting in exception from screenshot above.

Fix (to be reviewed by @bartekpacia ):
tldr: `scrollTo()` has to pass original finder or `finder.hitTestable().first`.
`scrollTo()` returns `finder.first` instead of original finder. So in this case it returned the stack widget, which is not hittestable. This was passed to `tap()` method and that is why the error message printed only this stack widget (`tap()` called `hitTestable()` on finder passed from `scrollTo()`, that's why it threw an error).

Another issue found in the process: https://github.com/leancodepl/patrol/issues/1048.

### PR DESCRIPTION
**Pull request by bartekpacia**
Fixes #1046

Huge thank you to @jBorkowska for such an in-depth investigation of the issue!

### CODE REVIEW

**Comment by bartekpacia** [comment]
This should also fix #1048.

**Review by jBorkowska — APPROVED** [review]

File: packages/patrol/lib/src/custom_finders/patrol_tester.dart [hunk 1]
```diff
@@ -439,7 +439,7 @@ class PatrolTester {
         settleTimeout: config.settleTimeout,
       );
 
-      return PatrolFinder(finder: finder.first, tester: this);
+      return PatrolFinder(finder: finder, tester: this);
```
Comment (bartekpacia) [inline packages/patrol/lib/src/custom_finders/patrol_tester.dart:442]: research if `finder.hitTestable()` breaks any existing tests

Comment (bartekpacia) [reply packages/patrol/lib/src/custom_finders/patrol_tester.dart:442]: do the same for liine 435

**Comment by bartekpacia** [comment]
Changing line 442 from:

```dart
return PatrolFinder(finder: finder, tester: this);
```

to:

```dart
return PatrolFinder(finder: finder.hitTestable(), tester: this);
```

doesn't break any tests. This is bad, because it means there's some behavior in our custom finders that is not covered by unit tests. I'll fix this.

**Comment by bartekpacia** [comment]
>  I'll fix this (add unit tests) in this PR.

Done.

**Review by jBorkowska — APPROVED** [review]

**Pull request merged**